### DISCUSSION TITLE
More Scipy interpolation methods for interp?

### DISCUSSION THREAD

**Discussion by meggart** [pydata/xarray#5055] [General] [answered]
Hi there, 

I am quite new to xarray and was wondering today if there is a specific reason why 'previous' and `next` are omitted from the available [scipy interpolation methods](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html#scipy.interpolate.interp1d) when using DataArray.interp. Maybe there is some other conical way in xarray to fill with previous/consecutive values? Or would it be worth making a PR that adds the additional methods to the method list (I just tried and it works....)

**Comment by dcherian** [comment, marked as answer]
I think those have been added after the xarray method was implemented. It *should* be easy to implement, maybe as simple as extending this list?
https://github.com/pydata/xarray/blob/fbd48d4307502f7dbe8afa37b84df59e74407820/xarray/core/missing.py#L455-L463

PRs welcome!

**Reply by meggart** [reply]
> maybe as simple as extending this list?

Yes, this is exactly what I did on my local copy and it worked. The only reason I was asking here was that it felt too simple to be true, So yes, I will prepare a PR.
